app/presenters/api/v1/results_presenter.rb

```diff
@@ -130,6 +131,19 @@ def sent
     current_user.shoutouts.where(time_period_id: time_period.id).filter_map { |shoutout| recipients_block(shoutout) }
   end
 
+  def public_shoutouts
+    shoutouts = Shoutout.where(time_period_id: time_period.id, public: true)
+    shoutouts.reject { |shoutout| current_user.shoutouts.include?(shoutout) }
+             .map { |shoutout| shoutout_block(shoutout) }
+  end
```

Comment:
A couple thoughts on this:
1. It might be more efficient and easier to read to use the `-` instead of parsing and rejecting. For example:
```ruby
Shoutout.where(...) - current_user.shoutouts
```
instead of 
```
Shoutout.where(...).reject{...}
```
2. I would separate out the presentation of the data, for example:
```
return_blocks = public_shoutouts.map{ |shoutout| shoutout_block(shoutout) }
```
because your method name (`public_shoutouts`) implies data and not how it's formatted.

3. If there is any chance of this being called more than once then please use memoization, e.g.:
```
def public_shoutouts
   @public_shoutouts ||= (
      ...
   )
end

Reply:
@rogergraves  can you explain, should I separate  within the method or create a separate method? 
`return_blocks = public_shoutouts.map{ |shoutout| shoutout_block(shoutout) }`

Reply:
Maybe just rename your methods:
 - `received` => `received_shoutout_blocks`
 - `sent` => `sent_shoutout_blocks`
 - `public_shoutouts` => `public_shoutout_blocks`

The idea is for the name of the method to describe what the method is returning, so that other developers don't have exam the method to see what is being returned.

Reply:
Thank you